__tests__/app.test.js

```diff
@@ -41,29 +41,106 @@ describe("/api/topics", () => {
+            return request(app).get('/api/comments').expect(200)
+        });
+        test('200: responds with an array of articles with enhanced properties', () => {
+            return request(app).get('/api/articles').expect(200)
             .then(({ body }) => {
                 const articles = body.articles
                 expect(Array.isArray(articles)).toBe(true)
-                expect(articles.length).toBe(1)
+                expect(articles.length).toBe(13)
                 articles.forEach(article => {
                     expect(article).toHaveProperty('author')
                     expect(article).toHaveProperty('title')
                     expect(article).toHaveProperty('article_id')
-                    expect(article).toHaveProperty('body')
                     expect(article).toHaveProperty('topic')
                     expect(article).toHaveProperty('created_at')
                     expect(article).toHaveProperty('votes')
                     expect(article).toHaveProperty('article_img_url')
-                });
+                    expect(article).toHaveProperty('comment_count')
+                })
+            })
+        });  
+        test('returns articles with no body property', () => {
+            return request(app).get('/api/articles')
+            .then(({ body }) => {
+                const articles = body.articles
+                articles.forEach((article) => {
+                    expect(article).not.toHaveProperty('body')
+                })
             })
         });
-        test('404: should return 404 when article does not exist', () => {
-            return request(app).get('/api/articles/999').expect(404)
+    });
+    describe('/api/articles/:article_id/comments ticket 6', () => {
+        test('200: responds with an array of comments for a given article_id', () => {
+            return request(app).get('/api/articles/1/comments').expect(200)
+            .then(({ body }) => {
+                const comments = body
+                expect(Array.isArray(comments)).toBe(true)
+                comments.forEach((comment) => {
+                    expect(comment).toHaveProperty('comment_id')
+                    expect(comment).toHaveProperty('votes')
+                    expect(comment).toHaveProperty('created_at')
+                    expect(comment).toHaveProperty('author')
+                    expect(comment).toHaveProperty('body')
+                    expect(comment).toHaveProperty('article_id')
+                })
+            })
+        });
+        test('404: should return 404 when article has not comments', () => {
+            return request(app).get('/api/articles/2/comments').expect(404)
             .then(({ body }) => {
                 const msg = body.msg
-                expect(msg).toBe('Article not found')
+                const errorMsg = 'No comments found for article with ID 2'
+                expect(msg).toBe(errorMsg)
+            })
```

Comment:
When the article has no comments, we would just want an empty array back rather than an error, if you imagine you write an article, as soon as you post it there will be no comments but nothing has gone wrong, unlike when you check for an article that doesn't exist in that case something has gone wrong.